lib/node_types/esp/src/device.h

```diff
@@ -120,51 +278,157 @@ class Subdevice {
+        
+        // IDENTIFICATION
+        Ustring name;  // Device name, used as MQTT topic component
+        
+        // PUBLISHING CONTROL
+        bool _ignore_case = true;        // Convert values to lowercase before comparison
+        bool _report_change = true;      // Should changes be published?
+        bool _needs_publishing = false;  // Does this device have unpublished changes?
+        bool _retained = false;          // Should MQTT messages be retained?
+
+        /**
+         * FILTER CALLBACK CHAIN
+         * =====================
+         * Called for every measurement to validate/modify the value.
+         * 
+         * The chain receives the device with measured_value filled in.
+         * Each callback can:
+         * - Modify measured_value
+         * - Return true to accept the value
+         * - Return false to reject the value
+         * 
+         * Common uses:
+         * - Range validation (reject out-of-range values)
+         * - Calibration (adjust readings)
+         * - Noise filtering (average multiple readings)
+         * - Unit conversion
+         * 
+         * Example:
+         *   device.add_filter([](Device& dev) {
+         *       float temp = dev.value().as_float();
+         *       if (temp < -40 || temp > 85) return false;  // Out of range
+         *       temp = temp * 1.02 - 0.5;  // Calibration
+         *       dev.value().from(temp);
+         *       return true;
+         *   });
+         */
         Callback_Link *filter_first;
 
-
-        // This is a chain of callbacks which are called based on
-        // a value change of the or one of any of the values of the subdevices.
-        // It gets passed the triggering device. 
-        // the last measured value will be in .value() or
-        // .value(n) with n denoting the subdevice.
-        // Returning false also allows here to discard the value.
-        // Returning true menas that value was considered.
-        // TODO: commented example
+        /**
+         * CHANGE CALLBACK CHAIN
+         * =====================
+         * Called when a value changes (after filter chain accepts it).
+         * 
+         * The chain receives the device with new measured_value.
+         * Each callback can:
+         * - Read the old value (last_confirmed_value)
+         * - Read the new value (measured_value)
+         * - Perform actions based on the change
+         * - Return true to accept the change
+         * - Return false to reject the change
+         * 
+         * Common uses:
+         * - Trigger actions (turn on light when motion detected)
+         * - Send alerts (notify when temperature too high)
+         * - Log changes
+         * - Update related devices
+         * 
+         * Example:
+         *   button.on_change([](Device& dev) {
+         *       if (dev.value().equals("on")) {
+         *           light.value().from("on");
```

Comment:
use the IN-Macro here to access aotehr defined device object - like IN(mylight).value().on();
(all outputs have an on function - better then using from)